src/apps/common/submodules/portWizard/portWizard.js

```diff
@@ -263,6 +268,175 @@ define(function(require) {
+				invalidMessageTemplate = self.getTemplate({
+					name: '!' + self.i18n.active().commonApp.portWizard.steps.nameAndNumbers.numbersToPort.errors.file,
+					data: {
+						type: file.type
+					}
+				}),
+				onParsedFile = function(results) {
+					var toastArgs;
+
+					if (results.numbersCount === 0) {
+						toastArgs = {
+							type: 'warning',
+							message: self.getTemplate({
+								name: '!' + self.i18n.active().commonApp.portWizard.steps.nameAndNumbers.numbersToPort.messages.file.noNumbers,
+								data: results
+							})
+						};
+					} else {
+						toastArgs = {
+							type: 'success',
+							message: self.getTemplate({
+								name: '!' + self.i18n.active().commonApp.portWizard.steps.nameAndNumbers.numbersToPort.messages.file.success,
+								data: results
+							})
+						};
+					}
+
+					monster.ui.toast(toastArgs);
+				},
+				parseFileArgs = _
+					.chain(args)
+					.pick('file', 'numbersArea')
+					.merge({
+						callback: onParsedFile
+					})
+					.value();
+
+			if (!isValid) {
+				monster.ui.toast({
+					type: 'error',
+					message: invalidMessageTemplate
+				});
+
+				return;
+			}
+
+			self.portWizardNameAndNumbersParseFile(parseFileArgs);
+
+			$fileNameInput.val(file.name);
+		},
+
+		/**
+		 * Parses the uploaded CSV file to extract the phone numbers
+		 * @param  {Object} args
+		 * @param  {Object} args.file  File data
+		 * @param  {jQuery} args.numbersArea  Text Area that contains the port request's phone numbers
+		 * @param  {Function} args.callback  Callback function to be executed once the parsing and
+		 *                                   number extraction has been completed
+		 */
+		portWizardNameAndNumbersParseFile: function(args) {
+			var self = this,
+				file = args.file,
+				$numbersArea = args.numbersArea,
+				callback = args.callback;
+
+			monster.waterfall([
+				function(waterfallCallback) {
+					Papa.parse(file, {
+						header: false,
+						skipEmptyLines: true,
+						complete: function(results) {
+							waterfallCallback(null, results);
+						}
+					});
+				},
+				function(results, waterfallCallback) {
+					var entries = _.flatten(results.data),
+						numbers = _
+							.chain(entries)
+							.map(_.trim)
+							.reject(_.isEmpty)
+							.map(monster.util.getFormatPhoneNumber)
+							.filter('isValid')
+							.map('e164Number')
+							.uniq()
+							.value();
+
+					waterfallCallback(null, entries, numbers);
+				},
+				function(entries, numbers, waterfallCallback) {
+					var countData = {
+						entriesCount: entries.length,
+						numbersCount: numbers.length
+					};
+
+					if (_.isEmpty(numbers)) {
+						waterfallCallback(null, countData);
+						return;
+					}
+
+					$numbersArea.val(function(i, text) {
+						var trimmedText = _.trim(text);
+
+						if (!(_.isEmpty(trimmedText) || _.endsWith(trimmedText, ','))) {
+							trimmedText += ', ';
+						}
+
+						return trimmedText + _.join(numbers, ', ');
+					}).focus();
```

Comment:
Not sure this logic belongs here as it is not really related to the file parsing.

What do you think of moving it to the event callback that triggered the file upload in the first place?

Reply:
I agree with you